File: MMORPGServer/GameServerApp/GameServerApp/Proto/WorldMap_CurrRoleResurgenceProto.cs
//===================================================
//作    者：边涯  http://www.u3dol.com  QQ群：87481002
//创建时间：2020-03-15 10:58:27
//备    注：
//===================================================
using System.Collections;
using System.Collections.Generic;
using System;

/// <summary>
/// 客户端发送角色复活消息
/// </summary>
public struct WorldMap_CurrRoleResurgenceProto : IProto
{
    public ushort ProtoCode { get { return 13015; } }

    public int Type; //复活类型

    public byte[] ToArray()
    {
        using (MMO_MemoryStream ms = new MMO_MemoryStream())
        {
            ms.WriteUShort(ProtoCode);
            ms.WriteInt(Type);
            return ms.ToArray();
        }
    }

    public static WorldMap_CurrRoleResurgenceProto GetProto(byte[] buffer)
    {
        WorldMap_CurrRoleResurgenceProto proto = new WorldMap_CurrRoleResurgenceProto();
        using (MMO_MemoryStream ms = new MMO_MemoryStream(buffer))
        {
            proto.Type = ms.ReadInt();
        }
        return proto;
    }
}

File: MMORPGServer/Proto/CreateLuaProto/WorldMap_CurrRoleResurgenceProto.lua
--客户端发送角色复活消息
WorldMap_CurrRoleResurgenceProto = { ProtoCode = 13015, Type = 0 }

--这句是重定义元表的索引，就是说有了这句，这个才是一个类
WorldMap_CurrRoleResurgenceProto.__index = WorldMap_CurrRoleResurgenceProto;

function WorldMap_CurrRoleResurgenceProto.New()
    local self = { }; --初始化self
    setmetatable(self, WorldMap_CurrRoleResurgenceProto); --将self的元表设定为Class
    return self;
end


--发送协议
function WorldMap_CurrRoleResurgenceProto.SendProto(proto)

    local ms = CS.LuaHelper.Instance:CreateMemoryStream();
    ms:WriteUShort(proto.ProtoCode);

    ms:WriteInt(proto.Type);

    CS.LuaHelper.Instance:SendProto(ms:ToArray());
    ms:Dispose();
end


--解析协议
function WorldMap_CurrRoleResurgenceProto.GetProto(buffer)

    local proto = WorldMap_CurrRoleResurgenceProto.New(); --实例化一个协议对象
    local ms = CS.LuaHelper.Instance:CreateMemoryStream(buffer);

    proto.Type = ms:ReadInt();

    ms:Dispose();
    return proto;
end